tests/integration/conftest.py

```diff
@@ -51,15 +54,37 @@ def pytest_collection_modifyitems(config, items):
 
 
 @pytest.fixture(scope="session", autouse=True)
-def logging_setup():
+def test_env_setup():
+    _logging_setup()
+    old_state = libnmstate.show()
+    _empty_net_state()
+    _ethx_init()
```

Comment:
If the operation fails/raises here, nothing will restore the old state and we will "loose" the tested node.

Reply:
I think you mentioned the pytest fixture does not need the `finally`, so the restore will always been invoked.

Reply:
OK. Maybe you mean after the `yield`.